File: testdata/jennies/builders/panel_builders/JavaBuilders/panelbuilder/PanelBuilder.java
package panelbuilder;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectWriter;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import java.util.List;
import dashboard.Panel;

public class PanelBuilder implements cog.Builder<Panel> {
    private Panel internal;

    public PanelBuilder() {
        this.internal = new Panel();
        this.onlyFromThisDashboard(false);
        this.onlyInTimeRange(false);
        this.limit(10);
        this.showUser(true);
        this.showTime(true);
        this.showTags(true);
        this.navigateToPanel(true);
        this.navigateBefore("10m");
        this.navigateAfter("10m");
    }
    public PanelBuilder onlyFromThisDashboard(Boolean onlyFromThisDashboard) {
    this.internal.onlyFromThisDashboard = onlyFromThisDashboard;
        return this;
    }
    public PanelBuilder onlyInTimeRange(Boolean onlyInTimeRange) {
    this.internal.onlyInTimeRange = onlyInTimeRange;
        return this;
    }
    public PanelBuilder tags(List<String> tags) {
    this.internal.tags = tags;
        return this;
    }
    public PanelBuilder limit(Integer limit) {
    this.internal.limit = limit;
        return this;
    }
    public PanelBuilder showUser(Boolean showUser) {
    this.internal.showUser = showUser;
        return this;
    }
    public PanelBuilder showTime(Boolean showTime) {
    this.internal.showTime = showTime;
        return this;
    }
    public PanelBuilder showTags(Boolean showTags) {
    this.internal.showTags = showTags;
        return this;
    }
    public PanelBuilder navigateToPanel(Boolean navigateToPanel) {
    this.internal.navigateToPanel = navigateToPanel;
        return this;
    }
    public PanelBuilder navigateBefore(String navigateBefore) {
    this.internal.navigateBefore = navigateBefore;
        return this;
    }
    public PanelBuilder navigateAfter(String navigateAfter) {
    this.internal.navigateAfter = navigateAfter;
        return this;
    }
    
    public Panel build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/panel_builders/RustBuilder/src/builders/panelbuilder.rs
use crate::cog;
use crate::types::panelbuilder;

#[derive(Debug, Clone)]
pub struct PanelBuilder {
    internal: panelbuilder::Panel,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl PanelBuilder {
    pub fn new() -> Self {
        Self {
            internal: panelbuilder::Panel::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for PanelBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl PanelBuilder {
    pub fn only_from_this_dashboard(mut self, only_from_this_dashboard: bool) -> Self {
        self.internal.only_from_this_dashboard = only_from_this_dashboard;

        self
    }
}

impl PanelBuilder {
    pub fn only_in_time_range(mut self, only_in_time_range: bool) -> Self {
        self.internal.only_in_time_range = only_in_time_range;

        self
    }
}

impl PanelBuilder {
    pub fn tags(mut self, tags: Vec<String>) -> Self {
        self.internal.tags = tags;

        self
    }
}

impl PanelBuilder {
    pub fn limit(mut self, limit: u32) -> Self {
        self.internal.limit = limit;

        self
    }
}

impl PanelBuilder {
    pub fn show_user(mut self, show_user: bool) -> Self {
        self.internal.show_user = show_user;

        self
    }
}

impl PanelBuilder {
    pub fn show_time(mut self, show_time: bool) -> Self {
        self.internal.show_time = show_time;

        self
    }
}

impl PanelBuilder {
    pub fn show_tags(mut self, show_tags: bool) -> Self {
        self.internal.show_tags = show_tags;

        self
    }
}

impl PanelBuilder {
    pub fn navigate_to_panel(mut self, navigate_to_panel: bool) -> Self {
        self.internal.navigate_to_panel = navigate_to_panel;

        self
    }
}

impl PanelBuilder {
    pub fn navigate_before(mut self, navigate_before: String) -> Self {
        self.internal.navigate_before = navigate_before;

        self
    }
}

impl PanelBuilder {
    pub fn navigate_after(mut self, navigate_after: String) -> Self {
        self.internal.navigate_after = navigate_after;

        self
    }
}

impl cog::Builder<panelbuilder::Panel> for PanelBuilder {
    fn build(&self) -> Result<panelbuilder::Panel, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
